media-proxy/tests/mesh_context_tests.cc

```diff
@@ -0,0 +1,51 @@
+#include <gtest/gtest.h>
+#include "mesh/concurrency.h"
+
+using namespace mesh;
+
+class test_class {
+  public:
+    context::Context _ctx;
+
+    void init(context::Context &ctx) { _ctx = context::WithCancel(ctx); }
+};
+
+/**
+ * Test designed to detect memory leaks in the Context class.
+ *
+ * Run the test with Valgrind to detect memory leaks.
+ * valgrind --leak-check=full ./media_proxy_unit_tests --gtest_filter=*Context*
+ */
+TEST(MeshContext, constructor) {
+    auto ctx = context::WithCancel(mesh::context::Background());
+    test_class t;
+    t.init(ctx);
+    {
+        auto ctx2 = context::WithCancel(ctx);
+        t.init(ctx2);
+        mesh::thread::Sleep(ctx2, std::chrono::milliseconds(10));
```

Comment:
I think this test case was supposed to verify the `context` implementation only. Maybe it's worth replacing `thread::Sleep` with `std::chrono::Sleep`?

Reply:
But in general, I'm okay with using `thread::Sleep`.

Reply:
Lets keep it as is